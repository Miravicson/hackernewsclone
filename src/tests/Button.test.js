import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import Button from "../components/Button"

describe("Test for the Button component", () => {
  const onClick = () => {}
  it("renders without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(<Button onClick={onClick}>Give Me More</Button>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  test("has a valid snapshot", () => {
    const compoonent = renderer.create(
      <Button onClick={onClick}>Give Me More</Button>
    )
    let tree = compoonent.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
