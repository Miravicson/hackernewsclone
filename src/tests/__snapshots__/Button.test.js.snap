// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the Button component has a valid snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Give Me More
</button>
`;
